coverage.py

```diff
@@ -754,12 +757,14 @@ def all(cls, _db, **kwargs):
         CollectionCoverageProvider (or, more likely, one of its subclasses).
 
         """
-        if not cls.PROTOCOL:
-            raise ValueError("%s must define PROTOCOL." % cls.__name__)
-        collections = _db.query(Collection).filter(
-            Collection.protocol==cls.PROTOCOL).order_by(func.random())
+        collections = _db.query(Collection)
+
+        if cls.PROTOCOL:
```

Comment:
I sense we may eventually need to let PROTOCOL be a list of protocols or perhaps have an `ANY_PROTOCOL = object()` constant, but it's not required for the MW right now.